taskchampion/taskchampion/src/server/sync/mod.rs

```diff
@@ -28,10 +29,11 @@ impl SyncServer {
     /// identify this client to the server.  Multiple replicas synchronizing the same task history
     /// should use the same client_id.
     pub fn new(origin: String, client_id: Uuid, encryption_secret: Vec<u8>) -> Result<SyncServer> {
+        let salt = dbg!(digest::digest(&digest::SHA256, client_id.as_ref()));
```

Comment:
What is the purpose of hashing the client_id rather than using it directly?

FYI, the question was brought to my attention when I noticed that the encryption.rs tests are still passing a Uuid to `Cryptor::new` without hashing it and that Uuid implements `AsRef<[u8]>`.

Reply:
I'm not sure what the point was, but I left it as-is here just to avoid unnecessarily introducing a breaking change.

Good point about the tests -- it doesn't hurt anything but would definitely cause some confusion for a reader. I'll fix it up.